modules/augment/src/flip.cpp

```diff
@@ -0,0 +1,43 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "precomp.hpp"
+namespace cv { namespace augment {
+
+FlipHorizontal::FlipHorizontal() {}
+
+void FlipHorizontal::image(InputArray src, OutputArray dst)
+{
+    cv::flip(src, dst, 1);
+}
+
+Point2f FlipHorizontal::point(const Point2f& src)
+{
+    return Point2f(srcImageCols - 1 - src.x, src.y);
+}
+
+Rect2f FlipHorizontal::rectangle(const Rect2f& src)
+{
+    return Rect2f(srcImageCols - 1 - (src.x + src.width), src.y, src.width, src.height);
```

Comment:
This is not correct implementation. In example, if `x == 0` and `width == 2`, it'll be transformed to `srcImageCols - 3` however it should be `srcImageCols - 2`.

Reply:
yes it shoulde be src.width-1 instead of src.width .. will fix it

Reply:
fixed